SU2_CFD/src/drivers/CDiscAdjMultizoneDriver.cpp

```diff
@@ -424,8 +431,18 @@ void CDiscAdjMultizoneDriver::SetRecording(unsigned short kind_recording, Kind_T
 
     for (iZone = 0; iZone < nZone; iZone++) {
 
+      /*--- In multi-physics, MESH_COORDS is an umbrella term for "geometric sensitivities",
+       *    if a zone has mesh deformation its recording type needs to change to MESH_DEFORM
+       *    as those sensitivities are managed by the adjoint mesh solver instead. ---*/
+
+      unsigned short type_recording = kind_recording;
+
+      if (Has_Deformation(iZone) && (kind_recording == MESH_COORDS)) {
+        type_recording = MESH_DEFORM;
+      }
+
       iteration_container[iZone][INST_0]->RegisterInput(solver_container, geometry_container,
-                                                        config_container, iZone, INST_0, kind_recording);
+                                                        config_container, iZone, INST_0, type_recording);
```

Comment:
To extract geometric sensitivities we need to do this kind of ugly `kind_recording` conversion for zones with mesh deformation...